collectors/googlestackdriver/collector.js

```diff
@@ -0,0 +1,183 @@
+        }
+        else {
+            endTs = moment(startTs).add(this.pollInterval, 'seconds').toISOString();
+        }
+        const resourceNames = JSON.parse(process.env.paws_collector_param_string_1);
+        const initialStates = resourceNames.map(resource => ({
+            resource,
+            nextPage:null,
+            since: startTs,
+            until: endTs,
+            poll_interval_sec: 1
+        }));
+        console.log('Inital states: ', initialStates);
+        return callback(null, initialStates, 1);
+    }
+    
+    pawsGetLogs(state, callback) {
+        let collector = this;
+
+        // Start API client
+        const client = new logging.v2.LoggingServiceV2Client({
+            credentials: JSON.parse(collector.secret)
+        });
+
+
+        console.info(`GSTA000001 Collecting data from ${state.since} till ${state.until} for ${state.resource}`);
+
+        // TODO: figure out a better way to format this. I'm pretty sure that it needs the newlines in it.
+        const filter = `timestamp >= "${state.since}"
+timestamp < "${state.until}"`;
+
+        let pagesRetireved = 0;
+        const options = {autoPaginate: false};
+
+        const paginationCallback = (result, acc = []) => {
+            console.log('getting page: ', pagesRetireved + 1, result[0].length);
+            pagesRetireved++;
+            const logs = result[0];
+            const nextPage = result[1];
+            const newAcc = [...acc, ...logs];
+            console.log('total Logs', newAcc.length);
+
+            if(nextPage && pagesRetireved < process.env.paws_max_pages_per_invocation){
+
+                return client.listLogEntries(nextPage, options)
+                    .then(res => paginationCallback(res, newAcc));
+            } else{
+                return {logs: newAcc, nextPage};
+            }
+        };
+
+        let params = state.nextPage ?
+            state.nextPage:
+            {
+                filter,
+                pageSize: 1000,
+                resourceNames:[state.resource]
+            };
+
+        client.listLogEntries(params, options)
+            .then(paginationCallback)
+            .then(({logs, nextPage}) => {
+                const newState = collector._getNextCollectionState(state, nextPage);
+                console.info(`GSTA000002 Next collection in ${newState.poll_interval_sec} seconds`);
+
+                return callback(null, logs, newState, newState.poll_interval_sec);
+            })
+            .catch(err => {
+                console.error(`GSTA000003 err in collection ${err}`);
+                
+                // Stackdriver Logging api has some rate limits that we might run into.
+                // If we run inot a rate limit error, instead of returning the error,
+                // we return the state back to the queue with an additional second added, up to 10
+                // https://cloud.google.com/logging/quotas
+                if(err.code === 8){
```

Comment:
please mention that 8 is a limit exceed error code. Or better use it as a global var, like
```
const API_THROTTLING_ERROR = 8;
```